### DISCUSSION TITLE
noneditable element can be deleted

### DISCUSSION THREAD

**Discussion by Tongyaoyu** [tinymce/tinymce#8816] [General]
In TinyMCE, if i add nonedit on some elements, it truely can not be editable, but it can be deleted. 
For example, 
```noneditable_class:` 'nonedit',<p class="nonedit">...apple orange...</p>```,
the cursor can be put in front of element p, then press Delete, the p element will be deleted, as well as 
```<table>...</table>```
I want to achieve the effect that the entire editor cannot be edited or deleted, and only the parts with class="editcontent"(editable_class: 'editcontent') added can be edited, similar to a large form that only the input can be edited.
I don't know how to achieve it. Can anyone help to anwser the question?

**Comment by lostkeys** [comment]
Hi @Tongyaoyu! Try use the brand new `editable_root: false` option. [Here is an example to get you started](https://fiddle.tiny.cloud/nAiaab)

**Reply by Tongyaoyu** [reply]
Thank you for your reply! I have encountered another problem. When I click the print button, there is always time and website address in the header and footer of the page. I want to remove them and customize my own header and footer, so what should I do？

**Reply by lostkeys** [reply]
TinyMCE does not do anything special around print, so it's using the browser default settings. I recommend you read up on CSS print styles and implement those in your own css which you'll load through `content_css` or `content_style` options.

**Reply by Tongyaoyu** [reply]
Thank you for your help，Best Wishes to you!
